internal/logging/logging.go

```diff
@@ -0,0 +1,197 @@
+		panic(err)
+	}
+	atom := zap.NewAtomicLevelAt(lvl)
+	enabled := atomic.NewBool(true)
+	encoder, syncer := selectEncoderSyncer(output)
+	logger := zap.New(zapcore.NewCore(
+		encoder,
+		syncer,
+		zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+			// Always log fatals, but allow other levels to be completely disabled.
+			return zap.FatalLevel.Enabled(lvl) || enabled.Load() && atom.Enabled(lvl)
+		}),
+	))
+
+	return &Logging{
+		enabled: enabled,
+		level:   atom,
+		output:  output,
+		logger:  logger.Sugar(),
+		mu:      sync.Mutex{},
+		loggers: map[string]*zap.SugaredLogger{},
+		options: options,
+	}
+}
+
+// GetChildLogger returns a named child sugared logger for the default logger `Logging.logger`.
+func (l *Logging) GetChildLogger(name string) *zap.SugaredLogger {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	// if the logger is already stored then use the stored child logger.
+	if logger, ok := l.loggers[name]; ok {
+		return logger
+	}
+
+	if level, found := l.options[name]; found {
+		var childlvl zapcore.Level
+		if err := childlvl.Set(level); err != nil {
+			panic(err)
+		}
+		atom := zap.NewAtomicLevelAt(childlvl)
+
+		encoder, syncer := selectEncoderSyncer(l.output)
+
+		// child logger cloned from default logger and using childlvl for LevelEnablerFunc.
+		logger := l.logger.Desugar().With(zap.Skip()).WithOptions(
+			zap.WrapCore(func(c zapcore.Core) zapcore.Core {
+				return zapcore.NewCore(
+					encoder,
+					syncer,
+					zap.LevelEnablerFunc(func(childlvl zapcore.Level) bool {
+						// Always log fatals, but allow other levels to be completely disabled.
+						return zap.FatalLevel.Enabled(childlvl) || l.enabled.Load() && atom.Enabled(childlvl)
+					}))
+			})).Sugar().Named(name)
+
+		l.loggers[name] = logger
+		return logger
+	} else {
+		logger := l.logger.Named(name)
+		l.loggers[name] = logger
+		return logger
+	}
+}
+
+// GetLogger returns the default sugared logger for the initialized Logging instance.
+func (l *Logging) GetLogger() *zap.SugaredLogger {
+	return l.logger
+}
+
+// selectEncoderSyncer returns Encoder and WriteSyncer for the logger based on output value.
+func selectEncoderSyncer(output string) (zapcore.Encoder, zapcore.WriteSyncer) {
+	switch output {
+	case "console":
+		return getConsoleEncoder(), getConsoleWriteSyncer()
+	case "systemd":
+		return getJsonEncoder(), getJournaldWriteSyncer()
+	default:
+		panic(fmt.Sprintf("%s is not a valid logger output", output))
+	}
+}
+
+// getConsoleEncoder returns console encoder used for console logging.
+func getConsoleEncoder() zapcore.Encoder {
+	return zapcore.NewConsoleEncoder(zap.NewDevelopmentEncoderConfig())
+}
+
+var defaultConfig = zap.Config{
+	Level: zap.NewAtomicLevelAt(zap.DebugLevel),
+
+	Development: true,
+	Sampling: &zap.SamplingConfig{
+		Initial:    100,
+		Thereafter: 100,
+	},
+
+	Encoding: "json",
+
+	// copied from "zap.NewProductionEncoderConfig" with some updates
+	EncoderConfig: zapcore.EncoderConfig{
```

Comment:
Only save `EncoderConfig` for now and don't disable `TimeKey`. You can override that in `NewJournalWriter` later. Make sure to pass the struct not a pointer to the struct!